internal/services/extendedlocation/extended_location_custom_location.go

```diff
@@ -0,0 +1,348 @@
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			if model := resp.Model; model != nil {
+				props := model.Properties
+
+				state := CustomLocationResourceModel{
+					Name:              id.CustomLocationName,
+					ResourceGroupName: id.ResourceGroupName,
+					Location:          model.Location,
+				}
+
+				if props != nil && props.Authentication != nil {
+					authType := pointer.From(props.Authentication.Type)
+					authValue := pointer.From(props.Authentication.Value)
+					state.Authentication = []AuthModel{
+						{
+							Type:  authType,
+							Value: authValue,
+						},
+					}
+				}
+
+				if props.ClusterExtensionIds != nil {
+					state.ClusterExtensionIds = *props.ClusterExtensionIds
+				}
+
+				if props.DisplayName != nil {
+					state.DisplayName = *props.DisplayName
+				}
+
+				if props.HostResourceId != nil {
+					state.HostResourceId = *props.HostResourceId
+				}
+
+				if props.HostType != nil {
+					state.HostType = string(*props.HostType)
+				}
+
+				if props.Namespace != nil {
+					state.Namespace = *props.Namespace
+				}
+
+				return metadata.Encode(&state)
+			}
+			return nil
+		},
+	}
+}
+
+func (r CustomLocationResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("deleting %s", *id)
+
+			if resp, err := client.Delete(ctx, *id); err != nil {
+				if !response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("deleting %s: %+v", *id, err)
+				}
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r CustomLocationResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state CustomLocationResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			customLocationProps := customlocations.CustomLocationProperties{}
+			d := metadata.ResourceData
+
+			if d.HasChanges("authentication") {
+				if state.Authentication != nil {
+					auth := state.Authentication[0]
+					customLocationProps.Authentication = &customlocations.CustomLocationPropertiesAuthentication{
+						Type:  &auth.Type,
+						Value: &auth.Value,
+					}
```

Comment:
I think we need to check the length before accessing the index 0, because it could be an empty block in the config.

Reply:
Will do.